b2g/components/RemoteControlService.jsm

```diff
@@ -158,8 +158,20 @@ this.RemoteControlService = {
   // 0000 and random generated number to a new string
   // Then get last 4 characters as new PIN code
   generatePIN: function() {
+    // Generate a random number in [0, 2^16 - 1]
+    let random = new Uint16Array(1);
+    let crypto = Services.wm.getMostRecentWindow("navigator:browser").crypto;
+
+    // Make sure this._pin exists and it's not zero
+    do {
+      crypto.getRandomValues(random);
+      // Get last needed digits of the random number
+      this._pin = random[0] % Math.pow(10, PINCODE_LENGTH);
+    } while(!this._pin);
```

Comment:
you might end up in an endless loop if there's an error. I think hard failing is better. Without randomness nothing will work.